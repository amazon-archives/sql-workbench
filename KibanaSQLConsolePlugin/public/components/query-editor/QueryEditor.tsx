import * as ace from 'brace';
import 'brace/mode/mysql';
import 'brace/mode/json';
import '../../ace-themes/sql_console';
import React from 'react';
import _ from 'lodash';
import { EuiPanel, EuiTitle, EuiButton, EuiText, EuiFlexGroup, EuiFlexItem, EuiCodeEditor } from '@elastic/eui';
import { ResponseDetail, TranslateResult } from '../main/main';

interface QueryEditorProps {
  onRun: (queriesString: string) => void;
  onTranslate: (queriesString: string) => void;
  onClear: () => void;
  queryTranslations: ResponseDetail<TranslateResult>[];
}

interface QueryEditorState {
  sqlQueriesString: string;
}

class QueryEditor extends React.Component<QueryEditorProps, QueryEditorState> {
  constructor(props: QueryEditorProps) {
    super(props);
    this.state = {
      sqlQueriesString: 'SELECT * FROM kibana_sample_data_ecommerce',
     // sqlQueriesString: this.props.sqlQueriesString,
    };

    this.updateSQLQueries = _.debounce(this.updateSQLQueries, 250).bind(this);
  }

  updateSQLQueries(newQueriesString: string) {
    this.setState({ sqlQueriesString: newQueriesString });
  }

  render() {
    return <EuiPanel className="sql-console-query-editor container-panel" paddingSize="none">
      <EuiTitle className="container-panel-header" size="l">
        <h1>SQL console</h1>
      </EuiTitle>
      <EuiFlexGroup gutterSize="none" className="resize-panel">
        <EuiFlexItem grow={1}>
          <EuiPanel className="sql-query-panel" paddingSize="none">
            <EuiText className="sql-query-panel-header">SQL</EuiText>
            <EuiCodeEditor
              mode="mysql"
              theme="sql_console"
              width="100%"
              value={this.state.sqlQueriesString}
              onChange={this.updateSQLQueries}
              showPrintMargin={false}
              setOptions={{
                fontSize: '12px',
                enableBasicAutocompletion: true,
                enableSnippets: true,
                enableLiveAutocompletion: true,
              }}
              onBlur={() => { console.log('blur'); }} // eslint-disable-line no-console
              aria-label="Code Editor"
            />
          </EuiPanel>
        </EuiFlexItem>
        <EuiFlexItem grow={1}>
          <EuiPanel className="translated-query-panel" paddingSize="none">
            <EuiText className="translated-query-panel-header">Elasticsearch query string</EuiText>
            <EuiCodeEditor
              mode="json"
              theme="sql_console"
              width="100%"
              value={this.props.queryTranslations.map((queryTranslation: any) =>
		                JSON.stringify(queryTranslation.data, null, 2)
		              ).join('\n')}
              onChange={() => {}}
              showPrintMargin={false}
              readOnly={true}
              setOptions={{
                fontSize: '12px',
                readOnly: true,
                highlightActiveLine: false,
                highlightGutterLine: false,
              }}
              onBlur={() => { console.log('blur'); }} // eslint-disable-line no-console
              aria-label="Code Editor"
            />
          </EuiPanel>
        </EuiFlexItem>
      </EuiFlexGroup>

      <div className="action-container">
        <EuiFlexGroup gutterSize="m">
          <EuiFlexItem grow={false} onClick={() => this.props.onRun(this.state.sqlQueriesString)}><EuiButton fill={true}>Run</EuiButton></EuiFlexItem>
          <EuiFlexItem grow={false} onClick={() => this.props.onTranslate(this.state.sqlQueriesString)}><EuiButton>Translate</EuiButton></EuiFlexItem>
          <EuiFlexItem grow={false} onClick={() => this.props.onClear()}><EuiButton>Clear</EuiButton></EuiFlexItem>
        </EuiFlexGroup>
      </div>
    </EuiPanel>;
  }
}

export default QueryEditor;
