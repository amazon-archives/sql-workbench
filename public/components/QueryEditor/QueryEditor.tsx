/*
 *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *   Licensed under the Apache License, Version 2.0 (the "License").
 *   You may not use this file except in compliance with the License.
 *   A copy of the License is located at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *   or in the "license" file accompanying this file. This file is distributed
 *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 *   express or implied. See the License for the specific language governing
 *   permissions and limitations under the License.
 */

import React from "react";
import _ from "lodash";
// @ts-ignore
import { EuiPanel, EuiTitle, EuiButton, EuiText, EuiFlexGroup, EuiFlexItem, EuiCodeEditor, EuiHeaderLink} from "@elastic/eui/lib";
import "brace/mode/mysql";
import "brace/mode/json";
import "../../ace-themes/sql_console";
import {ResponseDetail, TranslateResult} from "../Main/main";

interface QueryEditorProps {
  onRun: (queriesString: string) => void;
  onTranslate: (queriesString: string) => void;
  onClear: () => void;
  queryTranslations: ResponseDetail<TranslateResult>[];
  sqlQueriesString: string;
}

interface QueryEditorState {
  sqlQueriesString: string;
}

class QueryEditor extends React.Component<QueryEditorProps, QueryEditorState> {
  constructor(props: QueryEditorProps) {
    super(props);
    this.state = {
      sqlQueriesString: this.props.sqlQueriesString ? this.props.sqlQueriesString : ""
    };

    this.updateSQLQueries = _.debounce(this.updateSQLQueries, 250).bind(this);
  }

  updateSQLQueries(newQueriesString: string): void {
    this.setState({ sqlQueriesString: newQueriesString });
  }

  render() {
    return (
      <EuiPanel className="sql-console-query-editor container-panel"
        paddingSize="none"
      >
        <EuiTitle className="container-panel-header" size="l">
          <h1>SQL console</h1>
        </EuiTitle>

        <EuiFlexGroup gutterSize="none">
          <EuiFlexItem grow={1} className="sql-query-panel" paddingSize="none">
            <EuiText className="sql-query-panel-header">SQL Query</EuiText>
            <EuiCodeEditor
              mode="mysql"
              theme="sql_console"
              width="100%"
              height="18.5rem"
              value={this.state.sqlQueriesString}
              onChange={this.updateSQLQueries}
              showPrintMargin={false}
              setOptions={{
                fontSize: "12px",
                enableBasicAutocompletion: true,
                enableSnippets: true,
                enableLiveAutocompletion: true
              }}
              aria-label="Code Editor"
            />
          </EuiFlexItem>
          <EuiFlexItem
            grow={1}
            className="result-panel"
            paddingSize="none"
          >
            <EuiText className="translated-query-panel-header">
              Translation
            </EuiText>
            <EuiCodeEditor
              mode="json"
              theme="sql_console"
              width="100%"
              height="18.5rem"
              value={this.props.queryTranslations
                .map((queryResults: any) =>
                  JSON.stringify(queryResults.data, null, 2)
                )
                .join("\n")}
              showPrintMargin={false}
              readOnly={true}
              setOptions={{
                fontSize: "12px",
                readOnly: true,
                highlightActiveLine: false,
                highlightGutterLine: false
              }}
              aria-label="Code Editor"
            />
          </EuiFlexItem>
        </EuiFlexGroup>
        <div>
          <EuiFlexGroup className="action-container" gutterSize="m">
            <EuiFlexItem
              grow={false}
              onClick={() => this.props.onRun(this.state.sqlQueriesString)}
            >
              <EuiButton fill={true} className="sql-editor-button" >
                Run
              </EuiButton>
            </EuiFlexItem>
            <EuiFlexItem
              grow={false}
              onClick={() =>
                this.props.onTranslate(this.state.sqlQueriesString)
              }
            >
              <EuiButton className="sql-editor-button">
                Translate
              </EuiButton>
            </EuiFlexItem>
            <EuiFlexItem
              grow={false}
              onClick={() => {
                this.updateSQLQueries("");
                this.props.onClear();
              }}
            >
              <EuiHeaderLink className="sql-editor-link" href="#" isActive>
                Clear
              </EuiHeaderLink>
            </EuiFlexItem>
          </EuiFlexGroup>
        </div>
      </EuiPanel>
    );
  }
}

export default QueryEditor;
