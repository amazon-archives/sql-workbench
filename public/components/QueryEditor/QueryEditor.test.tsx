import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { render } from "@testing-library/react";
import QueryEditor from "./QueryEditor";


describe("<QueryEditor /> spec", () => {
  it("renders the component", () => {
    render(
      <QueryEditor
        onRun={() => {}}
        onTranslate={() => {}}
        onClear={() => {}}
        queryTranslations={[]}
      />
    );
    expect(document.body.children[0]).toMatchSnapshot();
  });
});