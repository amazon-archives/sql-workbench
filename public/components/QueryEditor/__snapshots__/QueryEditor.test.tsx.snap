// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QueryEditor /> spec renders the component 1`] = `
<div>
  <div
    class="euiPanel sql-console-query-editor container-panel"
  >
    <h1
      class="euiTitle euiTitle--large container-panel-header"
    >
      SQL console
    </h1>
    <div
      class="euiFlexGroup euiFlexGroup--directionRow euiFlexGroup--responsive"
    >
      <div
        class="euiFlexItem euiFlexItem--flexGrow1 sql-query-panel"
        paddingsize="none"
      >
        <div
          class="euiText euiText--medium sql-query-panel-header"
        >
          SQL
        </div>
        <div
          class="euiCodeEditorWrapper"
          data-test-subj="codeEditorContainer"
          style="width: 100%; height: 20em;"
        >
          <div
            class="euiCodeEditorKeyboardHint"
            data-test-subj="codeEditorHint"
            id="some_html_id"
            role="button"
            tabindex="0"
          >
            <p
              class="euiText"
            >
              Press Enter to start editing.
            </p>
            <p
              class="euiText"
            >
              When you're done, press Escape to stop editing.
            </p>
          </div>
          <div
            class=" ace_editor ace-sql-console"
            id="brace-editor"
            style="width: 100%; height: 20em; font-size: 12px;"
          >
            <textarea
              aria-label="Code Editor"
              autocapitalize="off"
              autocorrect="off"
              class="ace_text-input"
              spellcheck="false"
              style="opacity: 0;"
              tabindex="-1"
              wrap="off"
            />
            <div
              aria-hidden="true"
              class="ace_gutter"
            >
              <div
                class="ace_layer ace_gutter-layer ace_folding-enabled"
              />
              <div
                class="ace_gutter-active-line"
              />
            </div>
            <div
              class="ace_scroller"
            >
              <div
                class="ace_content"
              >
                <div
                  class="ace_layer ace_print-margin-layer"
                >
                  <div
                    class="ace_print-margin"
                    style="left: 4px; visibility: hidden;"
                  />
                </div>
                <div
                  class="ace_layer ace_marker-layer"
                />
                <div
                  class="ace_layer ace_text-layer"
                  style="padding: 0px 4px;"
                />
                <div
                  class="ace_layer ace_marker-layer"
                />
                <div
                  class="ace_layer ace_cursor-layer ace_hidden-cursors"
                >
                  <div
                    class="ace_cursor"
                  />
                </div>
              </div>
            </div>
            <div
              class="ace_scrollbar ace_scrollbar-v"
              style="display: none; width: 20px;"
            >
              <div
                class="ace_scrollbar-inner"
                style="width: 20px;"
              />
            </div>
            <div
              class="ace_scrollbar ace_scrollbar-h"
              style="display: none; height: 20px;"
            >
              <div
                class="ace_scrollbar-inner"
                style="height: 20px;"
              />
            </div>
            <div
              style="height: auto; width: auto; top: 0px; left: 0px; visibility: hidden; position: absolute; white-space: pre; overflow: hidden;"
            >
              <div
                style="height: auto; width: auto; top: 0px; left: 0px; visibility: hidden; position: absolute; white-space: pre; overflow: visible;"
              />
              <div
                style="height: auto; width: auto; top: 0px; left: 0px; visibility: hidden; position: absolute; white-space: pre; overflow: visible;"
              >
                XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="euiFlexItem euiFlexItem--flexGrow1 translated-query-panel"
        paddingsize="none"
      >
        <div
          class="euiText euiText--medium translated-query-panel-header"
        >
          Elasticsearch query string
        </div>
        <div
          class="euiCodeEditorWrapper"
          data-test-subj="codeEditorContainer"
          style="width: 100%; height: 20em;"
        >
          <div
            class="euiCodeEditorKeyboardHint"
            data-test-subj="codeEditorHint"
            id="some_html_id"
            role="button"
            tabindex="0"
          >
            <p
              class="euiText"
            >
              Press Enter to start editing.
            </p>
            <p
              class="euiText"
            >
              When you're done, press Escape to stop editing.
            </p>
          </div>
          <div
            class=" ace_editor ace-sql-console"
            id="brace-editor"
            style="width: 100%; height: 20em; font-size: 12px;"
          >
            <textarea
              aria-label="Code Editor"
              autocapitalize="off"
              autocorrect="off"
              class="ace_text-input"
              spellcheck="false"
              style="opacity: 0;"
              tabindex="-1"
              wrap="off"
            />
            <div
              aria-hidden="true"
              class="ace_gutter"
            >
              <div
                class="ace_layer ace_gutter-layer ace_folding-enabled"
              />
              <div
                class="ace_gutter-active-line"
                style="display: none;"
              />
            </div>
            <div
              class="ace_scroller"
            >
              <div
                class="ace_content"
              >
                <div
                  class="ace_layer ace_print-margin-layer"
                >
                  <div
                    class="ace_print-margin"
                    style="left: 4px; visibility: hidden;"
                  />
                </div>
                <div
                  class="ace_layer ace_marker-layer"
                />
                <div
                  class="ace_layer ace_text-layer"
                  style="padding: 0px 4px;"
                />
                <div
                  class="ace_layer ace_marker-layer"
                />
                <div
                  class="ace_layer ace_cursor-layer ace_hidden-cursors"
                >
                  <div
                    class="ace_cursor"
                  />
                </div>
              </div>
            </div>
            <div
              class="ace_scrollbar ace_scrollbar-v"
              style="display: none; width: 20px;"
            >
              <div
                class="ace_scrollbar-inner"
                style="width: 20px;"
              />
            </div>
            <div
              class="ace_scrollbar ace_scrollbar-h"
              style="display: none; height: 20px;"
            >
              <div
                class="ace_scrollbar-inner"
                style="height: 20px;"
              />
            </div>
            <div
              style="height: auto; width: auto; top: 0px; left: 0px; visibility: hidden; position: absolute; white-space: pre; overflow: hidden;"
            >
              <div
                style="height: auto; width: auto; top: 0px; left: 0px; visibility: hidden; position: absolute; white-space: pre; overflow: visible;"
              />
              <div
                style="height: auto; width: auto; top: 0px; left: 0px; visibility: hidden; position: absolute; white-space: pre; overflow: visible;"
              >
                XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="action-container"
    >
      <div
        class="euiFlexGroup euiFlexGroup--gutterMedium euiFlexGroup--directionRow euiFlexGroup--responsive action-container"
      >
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        >
          <button
            class="euiButton euiButton--primary sql-editor-button euiButton--fill"
            type="button"
          >
            <span
              class="euiButton__content"
            >
              <span
                class="euiButton__text"
              >
                Run
              </span>
            </span>
          </button>
        </div>
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        >
          <button
            class="euiButton euiButton--primary sql-editor-button"
            type="button"
          >
            <span
              class="euiButton__content"
            >
              <span
                class="euiButton__text"
              >
                Translate
              </span>
            </span>
          </button>
        </div>
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        >
          <a
            class="euiButtonEmpty euiButtonEmpty--primary euiHeaderLink sql-editor-link"
            href="#"
          >
            <span
              class="euiButtonEmpty__content"
            >
              <span
                class="euiButtonEmpty__text"
              >
                Clear
              </span>
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
